solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/outbound/JCSMPOutboundMessageHandler.java

```diff
@@ -96,28 +102,64 @@ public void handleMessage(Message<?> message) throws MessagingException {
 					String.format("Unable to parse header %s", SolaceBinderHeaders.CONFIRM_CORRELATION), e);
 		}
 
-		XMLMessage xmlMessage = xmlMessageMapper.map(message, properties.getExtension().getHeaderExclusions(),
-				properties.getExtension().isNonserializableHeaderConvertToString());
-		correlationKey.setRawMessage(xmlMessage);
-		xmlMessage.setCorrelationKey(correlationKey);
+		List<XMLMessage> smfMessages;
+		Object proxyCorrelationKey;
+		if (message.getHeaders().containsKey(SolaceBinderHeaders.BATCHED_HEADERS)) {
+			LOGGER.debug("Detected header {}, handling as batched message (Message<List<?>>) <message handler ID: {}>",
+					SolaceBinderHeaders.BATCHED_HEADERS, id);
+			smfMessages = xmlMessageMapper.mapBatchMessage(
+					message,
+					properties.getExtension().getHeaderExclusions(),
+					properties.getExtension().isNonserializableHeaderConvertToString());
 
-		LOGGER.debug("Publishing message to destination [ {}:{} ]",
-				targetDestination instanceof Topic ? "TOPIC" : "QUEUE", targetDestination);
+			proxyCorrelationKey = transactedSession != null ?
```

Comment:
for transacted batch BatchProxyCorrelationKey is expected to be used not the original correlationKey?

Reply:
It's unnecessary in this case, because we're resolving the correlation key manually in this handler after commit or rollback in the case of `transacted=true`.

Reply:
resolution code from below, it happens using proxy which does not have a wrapped original correlation key?

				....
producerEventHandler.responseReceivedEx(proxyCorrelationKey);

Reply:
Fixed.